blueoil/utils/tfds_builders/segmentation.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+import numpy as np
+import tensorflow_datasets as tfds
+
+
+class SegmentationBuilder(tfds.core.GeneratorBasedBuilder):
+    """
+    A custom TFDS builder for segmentation dataset.
+    This class loads data from existing dataset classes and
+    generate TFDS formatted dataset which is equivalent to the original one.
+    See also: https://www.tensorflow.org/datasets/add_dataset
+    """
+
+    VERSION = tfds.core.Version("0.1.0")
+
+    def __init__(self, dataset_name, dataset_class=None, dataset_kwargs=None, **kwargs):
+        self.name = dataset_name
+        self.dataset_class = dataset_class
+        self.dataset_kwargs = dataset_kwargs
+        super().__init__(**kwargs)
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description="Custom TFDS dataset for segmentation",
+            features=tfds.features.FeaturesDict({
+                "image": tfds.features.Image(),
+                "label": tfds.features.ClassLabel(),
```

Comment:
So the feature `label` is used for having a list of classes but not needed for every example. Is my understanding correct?

Reply:
@tfujiwar 
Yes! Your understanding is correct! 👍

Reply:
Thank you!